packages/lisk-p2p/src/response_handler.ts

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright © 2018 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ *
+ */
+import { isAlpha, isIP, isNumeric, isPort } from 'validator';
+import { InvalidPeerAddress } from './errors';
+import { PeerConfig } from './peer';
+
+const IPV4_NUMBER = 4;
+const IPV6_NUMBER = 6;
+
+interface RawPeerObject {
+	readonly height?: string;
+	readonly ip: string;
+	readonly os?: string;
+	readonly version?: string;
+	readonly wsPort: string;
+}
+
+export const checkPeerAddress = (peer: unknown): boolean => {
+	if (!peer) {
+		return false;
+	}
+
+	const { ip, wsPort } = peer as RawPeerObject;
+
+	if (!ip || !wsPort) {
+		return false;
+	}
+
+	if ((!isIP(ip, IPV4_NUMBER) && !isIP(ip, IPV6_NUMBER)) || !isPort(wsPort)) {
+		return false;
+	}
+
+	return true;
+};
+
+export const instantiatePeerFromResponse = (
+	peer: unknown,
+): PeerConfig | Error => {
```

Comment:
it's not returning error anywhere?

Reply:
it is returning at [response_handler.ts#L52](https://github.com/LiskHQ/lisk-elements/blob/062056245576654306d35340c94e58722534d334/packages/lisk-p2p/src/response_handler.ts#L52)

Reply:
Discussed with @jondubois to throw an error instead of returning an error to have consistency across the p2p lib